lib/matplotlib/axes/_axes.py

```diff
@@ -7234,6 +7241,16 @@ def stairs(self, values, edges=None, *,
                                    fill=fill,
                                    **kwargs)
         self.add_patch(patch)
+        if baseline is None and fill:
+            _api.warn_external(
+                f"Both {baseline=} and {fill=} have been passed. "
+                "baseline=None is only intended for unfilled stair plots. "
+                "Because baseline is None, the Path used to draw the stairs will "
+                "not be closed, thus because fill is True the polygon will be closed "
+                "by drawing an (unstroked) edge from the first to last point.  It is "
```

Comment:
I find this a bit difficult to parse, as there are two 'because' in the sentence, and also two conjunctions (thus because) in the second half.